Modules/_localemodule.c

```diff
@@ -269,7 +274,12 @@ PyLocale_strxfrm(PyObject* self, PyObject* args)
             goto exit;
         }
         buf = new_buf;
+        errno = 0;
         n2 = wcsxfrm(buf, s, n2+1);
+        if (errno) {
+            PyErr_SetFromErrno(PyExc_OSError);
+            goto exit;
+        }
     }
     result = PyUnicode_FromWideChar(buf, n2);
 exit:
```

Comment:
Except adding an overhead for function call.

But I don't think it matters in comparison with the time of calling other functions.